apps/passport-client/pages/index.tsx

```diff
@@ -115,10 +126,40 @@ class App extends React.Component<object, AppState> {
   }
 
   startBackgroundJobs = () => {
-    console.log("Starting background jobs...");
+    console.log("[JOB] Starting background jobs...");
     this.jobPollUser();
+    this.startJobSyncOfflineCheckins();
+    this.jobCheckConnectivity();
   };
 
+  jobCheckConnectivity = async () => {
+    window.addEventListener("offline", () => this.setIsOffline(true));
+    window.addEventListener("online", () => this.setIsOffline(false));
+  };
+
+  setIsOffline(offline: boolean) {
+    console.log(`[CONNECTIVITY] ${offline ? "offline" : "online"}`);
+    this.update({
+      ...this.state,
+      offline: offline
+    });
```

Comment:
Normally we would do something like this by calling `dispatch()`, rather than updating the state directly. Is there a reason for doing it differently?